File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer =  0;
        while(true){
            if(a%2 == 0){
                if(a-1 == b){
                    answer++;
                    break;
                }
                else{
                    a = a/2;
                    if(b%2 ==0){
                        b = b/2;
                    }
                    else b = b/2+1;
                    answer++;
                }
            }
            else if(a%2 != 0){
                if(a+1 == b){
                    answer++;
                    break;
                }
                else{
                    a = a/2+1;
                    if(b%2 ==0){
                        b = b/2;
                    }
                    else b = b/2+1;
                    answer++;
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.js
function solution(n,a,b) {
    let cnt = 1;
    
    while (true) {
        if (Math.floor((a + 1) / 2) === Math.floor((b + 1) / 2)) break;
        
        a = (a + 1) / 2;
        b = (b + 1) / 2;
        cnt++;
    }
    return cnt;
}